wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -459,6 +469,26 @@ private void validateInputSchema(@Nullable Schema inputSchema, FailureCollector
     }
   }
 
+  private void validatePrecondition(String precondition, String field, FailureCollector collector) {
+    if (precondition == null) {
+      collector.addFailure(String.format("Precondition %s must be present.", field.substring(12)).toUpperCase(),
+                           null)
+        .withConfigProperty(field);
+    }
+  }
+
+  private void validateDirectivesAndUDDs(FailureCollector collector) {
```

Comment:
UDDs are also technically directives so I think we can name this `validateDirectives()` to simplify it.